pkg/util/block_source_reader.go

```diff
@@ -0,0 +1,138 @@
+package util
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"io"
+	"sync"
+)
+
+/*
+ A BlockSource is a data source that reads/writes in block size increments.
+ */
+type BlockSource interface {
+	Read(startSector uint64, numSectors uint64, buffer []byte) (sectorsRead uint64, err error)
+	Capacity() int64
+	BlockSize() int
```

Comment:
What is reason one of the two funcs return int64 while another one returns int?

Reply:
We're still at 4KB blocks for most devices, up to 2GB blocks seems big enough